### PR TITLE
Slider labelStepSize=0 hides axis labels

### LINKED ISSUE #828: labelStepSize in slider can take 0 and crash browser
### Bug report

- __Package version(s)__: 1.11.1
- __Browser and OS versions__:  Chrome 57.0.2987.98 (64-bit) on ArchLinux x64

#### Steps to reproduce

1. Set the `labelStepSize` of a slider to 0
1. Try it in browser

#### Actual behavior

It does not perform validation on this prop.
This will crash the chrome process, showing `Aw, Snap`.

Since it simply loops by doing: 
```js
 for (let i = min, left = 0; i < max || approxEqual(i, max); i += labelStepSize, left += stepSize) {
            labels.push(<div className={`${Classes.SLIDER}-label`} key={i} style={{left}}>{this.formatLabel(i)}</div>);
        }
```

#### Expected behavior

At least `assert(labelStepSize > 0)` before looping. A better implementation would be not to show labels at all

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

passing `labelStepSize={0}` will hide all axis labels (like they're infinitely close together and they all got sucked into a black hole ⚫️).
all labels can be disabled entirely with `renderLabel={false}`, but this option allows for handle labels only!

#### Reviewers should focus on:

is this weird? or is it cool?

#### Screenshot

image
^ note the lack of axis labels on the third one 🙌

### CODE REVIEW

File: packages/core/test/slider/sliderTests.tsx [hunk 1]
```diff
@@ -43,6 +43,14 @@ describe("<Slider>", () => {
         assert.lengthOf(wrapper.find(`.${Classes.SLIDER}-label`), 7);
     });
 
+    it("labelStepSize=0 renders only min and max axis labels", () => {
+        const wrapper = renderSlider(<Slider min={10} max={20} labelStepSize={0}/>);
+        const labels = wrapper.find(`.${Classes.SLIDER}-label`);
+        assert.lengthOf(labels, 3);
```
Comment (leebyp) [inline packages/core/test/slider/sliderTests.tsx]: look for the label inside the `pt-slider-axis`  instead? it's odd that the description is for the min, max but the test is also checking for the label on the handle

File: packages/core/test/slider/sliderTests.tsx [hunk 2]
```diff
@@ -43,6 +43,14 @@ describe("<Slider>", () => {
         assert.lengthOf(wrapper.find(`.${Classes.SLIDER}-label`), 7);
     });
 
+    it("labelStepSize=0 renders only min and max axis labels", () => {
+        const wrapper = renderSlider(<Slider min={10} max={20} labelStepSize={0}/>);
+        const labels = wrapper.find(`.${Classes.SLIDER}-label`);
```
Comment (leebyp) [inline packages/core/test/slider/sliderTests.tsx]: we have this full class in the `.ts` file

File: packages/core/src/components/slider/coreSlider.tsx [hunk 3]
```diff
@@ -124,8 +125,11 @@ export abstract class CoreSlider<P extends ICoreSliderProps> extends AbstractCom
     }
 
     private maybeRenderAxis() {
-        const { max, min, labelStepSize } = this.props;
-        if (this.props.renderLabel === false) { return undefined; }
+        const { max, min } = this.props;
+        const labelStepSize = this.props.labelStepSize || (max - min);
```
Comment (leebyp) [inline packages/core/src/components/slider/coreSlider.tsx]: are we still safe if `labelStepSize === 0 && max === min`?
also, i thought we're avoiding boolean coercion?

**Comment by leebyp** [comment]
on the feature itself, yeah i like the short hand as long as we put something in the docs since it's an opt in feature and the logic jump isn't big

File: packages/core/test/slider/sliderTests.tsx (hunk 2 above)
Comment (giladgray) [inline packages/core/test/slider/sliderTests.tsx]: ah yeah i figured that out later

File: packages/core/src/components/slider/coreSlider.tsx (hunk 3 above)
Comment (giladgray) [inline packages/core/src/components/slider/coreSlider.tsx]: ah yeah good call on both counts

**Comment by leebyp** [comment]
cool with me, but could do with a 3rd look on the assumed API

**Comment by giladgray** [comment]
💡 found a better use for `0` that's less magical: it simply hides all axis labels. but because `renderLabel` is (or may be) defined, handle labels can still appear. it's a pretty cool hybrid. the next preview demonstrates it in the third example slider.

**Comment by leebyp** [comment]
oooo nice, i think there's a pretty good design case of just having the start and end that we had - can we add that as an additional example / edit the second one?

**Comment by giladgray** [comment]
i don't think we need to demonstrate the min/max usage--there's nothing special there, it's just math.

**Comment by giladgray** [comment]
starting over with errors on 0 instead of this madness

**Pull request closed without merging**